### PR TITLE
Porting Teaser, OperationalHours components from EEC.

### PR DESCRIPTION
**Pull request by smurrayatwork**
Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)

## Description
This ports over the OperationalHours and Teaser components from the EEC build.

## Steps to Test
1.  Pull the branch. OperationalHours is located under atoms/time/OperationalHours and Teaser is under organisms.

### CODE REVIEW

File: react/src/components/organisms/Teaser/index.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  handleMouseLeave = () => {
+    this.setState({ isActive: false });
+    if (typeof this.props.clearActiveHover === 'function') {
+      this.props.clearActiveHover()
+    }
+  };
+
+  render() {
+    const primaryInfoClass =
```
Comment (avrilpearl) [inline react/src/components/organisms/Teaser/index.js]: can you use classnames library here - see button atoms for example

File: react/src/components/organisms/Teaser/index.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      this.props.left && !this.props.right
+        ? 'ma__general-teaser__primaryinfo__details'
+        : 'ma__general-teaser__primaryinfo';
+    const secondaryInfoClass =
```
Comment (avrilpearl) [inline react/src/components/organisms/Teaser/index.js]: use class names here

File: react/src/components/organisms/Teaser/storyutils.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
+import React from 'react';
```
Comment (avrilpearl) [inline react/src/components/organisms/Teaser/storyutils.js:1]: should any of these be pulled out into a higher utils folder?

File: react/src/components/organisms/Teaser/style.scss [hunk 4]
```diff
@@ -0,0 +1,285 @@
+@import "../../styles/global";
```
Comment (avrilpearl) [inline react/src/components/organisms/Teaser/style.scss]: move this style.scss for teaser to the assets folder level and import it here

File: react/src/index.js [hunk 5]
```diff
@@ -4,6 +4,8 @@
 */
 
 //@base
+import OperationalHours from "./components/atoms/time/OperationalHours";
```
Comment (avrilpearl) [inline react/src/index.js]: why are you importing here?

File: react/src/components/atoms/time/OperationalHours/style.scss [hunk 6]
```diff
@@ -0,0 +1,11 @@
+.ma__operational-hours {
```
Comment (avrilpearl) [inline react/src/components/atoms/time/OperationalHours/style.scss]: move this to the assets folder and import here.

File: react/src/components/organisms/Teaser/storyutils.js (hunk 3 above)
Comment (smurrayatwork) [reply react/src/components/organisms/Teaser/storyutils.js:1]: In this case I want to say no; If while working on another component you find you need them, then they should be pulled out, but not before. A lot of the options are pretty specific to the story for this Teaser component.

File: react/src/index.js (hunk 5 above)
Comment (smurrayatwork) [reply react/src/index.js]: Bug from my IDE

File: assets/scss/03-organisms/_teaser.scss [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+    flex-flow: column;
+    flex: 1 50%;
+
+    &--border {
+      flex: 1 1 40%;
+      flex-grow: 1;
+      border-left-style: solid;
+      border-left-width: 1px;
+      padding-left: 0.75em;
+      border-left-color: $c-gray-dcdcdc;
```
Comment (avrilpearl) [inline assets/scss/03-organisms/_teaser.scss]: can you update this to `$c-gray-light`

File: assets/scss/03-organisms/_teaser.scss [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+
+  &__secondary {
+    display: inline-flex;
+    flex-flow: column;
+    flex: 1 50%;
+
+    &--border {
+      flex: 1 1 40%;
+      flex-grow: 1;
+      border-left-style: solid;
```
Comment (avrilpearl) [inline assets/scss/03-organisms/_teaser.scss]: simplify all here to --> `border-left: 1px solid $c-gray-light;`

File: assets/scss/03-organisms/_teaser.scss [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+      border-left-width: 1px;
+      padding-left: 0.75em;
+      border-left-color: $c-gray-dcdcdc;
+    }
+
+    @media ($bp-small-min) {
+      flex: 1 1 40%;
+      flex-grow: 1;
+      border-left-style: none;
```
Comment (avrilpearl) [inline assets/scss/03-organisms/_teaser.scss]: simplify this and line below to `border-left: none;`

File: assets/scss/03-organisms/_teaser.scss [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+      margin-left: 0.25em;
+    }
+  }
+
+  &__secondaryicon svg {
+    display: inline-block;
+    vertical-align: middle;
+    height: 1.1rem;
+    width: 1.1rem;
+    fill: $c-theme-green;
```
Comment (avrilpearl) [inline assets/scss/03-organisms/_teaser.scss]: update this from `$c-theme-green` to `$c-primary-alt`

File: assets/scss/03-organisms/_teaser.scss [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+  }
+  .ma__general-teaser__primary {
+    flex-grow: 2;
+    &__details {
+
+    }
+  }
+  .ma-icon{
+    &__green {
+      fill: $c-theme-green;
```
Comment (avrilpearl) [inline assets/scss/03-organisms/_teaser.scss]: update to `$c-primary-alt`

File: assets/scss/03-organisms/_teaser.scss [hunk 12]
```diff
@@ -0,0 +1,288 @@
...
+  svg.link-arrow {
+    height: 1rem;
+    width: 1rem;
+    fill: $c-bay-blue-lighter;
+  }
+  .ma__decorative-link {
+    svg {
+      height: 1rem;
+      width: 1rem;
+      fill: $c-bay-blue-lighter;
```
Comment (avrilpearl) [inline assets/scss/03-organisms/_teaser.scss]: update to `$c-primary-lighter`

File: assets/scss/03-organisms/_teaser.scss [hunk 13]
```diff
@@ -0,0 +1,288 @@
...
+    display: inline-block;
+    vertical-align: middle;
+    height: 1.1rem;
+    width: 1.1rem;
+    fill: $c-theme-green;
+  }
+  svg.link-arrow {
+    height: 1rem;
+    width: 1rem;
+    fill: $c-bay-blue-lighter;
```
Comment (avrilpearl) [inline assets/scss/03-organisms/_teaser.scss]: update to `$c-primary-lighter`

File: assets/scss/03-organisms/_teaser.scss [hunk 14]
```diff
@@ -0,0 +1,288 @@
...
+    line-height: 27px;
+    a {
+      font-weight: 500;
+    }
+    svg {
+      height: 1rem;
+      width: 1rem;
+      margin-right: 8px;
+    }
+    &__number {
```
Comment (avrilpearl) [inline assets/scss/03-organisms/_teaser.scss]: nothing in here so you can get rid of it

File: assets/scss/03-organisms/_teaser.scss [hunk 15]
```diff
@@ -0,0 +1,288 @@
...
+    border-left-color: $c-bd-divider;
+    &__show-more {
+      &.show-fewer {
+        padding-left: 1rem;
+        svg {
+          transform: rotate(180deg);
+        }
+      }
+      svg {
+        fill: rgba(20, 85, 143, 0.5);
```
Comment (avrilpearl) [inline assets/scss/03-organisms/_teaser.scss]: update file to rgba($c-primary, 0.5)

File: assets/scss/03-organisms/_teaser.scss [hunk 16]
```diff
@@ -0,0 +1,288 @@
...
+    font-size: 18px;
+    line-height: 1;
+    margin-bottom: 1em;
+    color: $c-font-dark;
+    font-style: italic;
+    font-weight: 700;
+  }
+
+  &__date + &__org {
+    border-left-style: solid;
```
Comment (avrilpearl) [inline assets/scss/03-organisms/_teaser.scss]: update to `border-left: 1px solid;`

File: assets/scss/03-organisms/_teaser.scss [hunk 17]
```diff
@@ -0,0 +1,288 @@
...
+        font-size: 0.95rem;
+      }
+      &__item {
+        font-size: 1.24rem;
+      }
+      &__name {
+        margin-bottom: 0;
+        span {
+          font-size: 1.25rem;
+          color: $c-theme-gray;
```
Comment (avrilpearl) [inline assets/scss/03-organisms/_teaser.scss]: update to `$c-gray-dark`

File: assets/scss/03-organisms/_teaser.scss [hunk 18]
```diff
@@ -0,0 +1,288 @@
+@import "../00-base/variables";
+@import "../00-base/colors";
+@import "../00-base/breakpoints";
+
+
+$c-theme-gray: #535353;
```
Comment (avrilpearl) [inline assets/scss/03-organisms/_teaser.scss]: delete these variable

File: assets/scss/03-organisms/_teaser.scss [hunk 19]
```diff
@@ -0,0 +1,288 @@
+@import "../00-base/variables";
```
Comment (avrilpearl) [inline assets/scss/03-organisms/_teaser.scss]: you should not need to import any of these at the assets level, i.e. you can delete these

File: react/src/components/atoms/time/OperationalHours/index.js [hunk 20]
```diff
@@ -0,0 +1,97 @@
...
+                </tr>
+              );
+            })}
+        </tbody>
+      </table>
+    </div>
+  );
+};
+
+OperationalHours.propTypes = {
```
Comment (avrilpearl) [inline react/src/components/atoms/time/OperationalHours/index.js:62]: can you add comments above these props to describe their use.

File: react/src/components/organisms/Teaser/index.js [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+            </div>
+          )}
+        </div>
+      </section>
+    );
+  }
+}
+
+Teaser.propTypes = {
+  id: PropTypes.number,
```
Comment (avrilpearl) [inline react/src/components/organisms/Teaser/index.js:86]: can you add comments to these props about usage

Comment (mark-casias) [reply react/src/components/organisms/Teaser/index.js:86]: I am not sure, but don't you need to use the /** */ pattern so the description is picked up in the information section?

Comment (avrilpearl) [reply react/src/components/organisms/Teaser/index.js:86]: yep @smurrayatwork

**Review by mark-casias — APPROVED** [review]
Works well in the browser. There are no knobs for the teasers, not sure if that's a deal breaker or not, so I approve.

File: assets/scss/03-organisms/_teaser.scss [hunk 22]
```diff
@@ -0,0 +1,277 @@
+@import "../00-base/colors";
+@import "../00-base/breakpoints";
+
+$c-gray-dcdcdc: #dcdcdc;
+$c-theme-green: #43956f;
+
+.ma__results-list-container{
+  .ma__general-teaser {
```
Comment (clairesunstudio) [inline assets/scss/03-organisms/_teaser.scss:8]: what's the relationship between Teaser and General Teaser component? Seems like they are using the same class names

File: react/src/components/organisms/Teaser/index.js [hunk 23]
```diff
@@ -0,0 +1,118 @@
...
+      ]);
+    }
+  },
+  /** An array of components to be displayed under the title, on the right.
+      If left is not set, this will display on the left side under the title instead.
+      Components that may be passed in the array: DecorativeLink, OperationalHours, IconLink. */
+  right: (props, propName, componentName) => {
+    if (props[propName]) {
+      return componentArrayPropTypeCheck(props, propName, componentName, [
+        'DecorativeLink',
```
Comment (clairesunstudio) [inline react/src/components/organisms/Teaser/index.js:110]: what's the usage difference here Link vs DecorativeLink vs Icon Link

File: assets/scss/03-organisms/_teaser.scss (hunk 22 above)
Comment (smurrayatwork) [reply assets/scss/03-organisms/_teaser.scss:8]: They just have similar styling needs in this case.

File: react/src/components/organisms/Teaser/index.js (hunk 23 above)
Comment (smurrayatwork) [reply react/src/components/organisms/Teaser/index.js:110]: In this case, any of those could be added for support. On EEC we needed it to be DecorativeLink.

**Review by clairesunstudio — APPROVED** [review]
approved for Teaser import, follow up ticket to merge General Teaser into Teaser --> https://github.com/massgov/budget/pull/6

**Pull request merged**